app/lib/import_export.rb

```diff
@@ -692,222 +368,62 @@ def extract_assessments_details(patients_identifiers, assessments, fields)
     assessments_details
   end
 
-  # Extracts laboratory data values given relevant fields
+  # Extract laboratory data values given relevant fields
   def extract_laboratories_details(patients_identifiers, laboratories, fields)
     laboratories.map { |laboratory| laboratory.custom_details(fields, patients_identifiers[laboratory.patient_id]) }
   end
 
-  # Extracts close contact data values given relevant fields
+  # Extract close contact data values given relevant fields
   def extract_close_contacts_details(patients_identifiers, close_contacts, fields)
     close_contacts.map { |close_contact| close_contact.custom_details(fields, patients_identifiers[close_contact.patient_id]) }
   end
 
-  # Extracts transfer data values given relevant fields
+  # Extract transfer data values given relevant fields
   def extract_transfers_details(patients_identifiers, transfers, fields)
-    user_emails = Hash[User.find(transfers.map(&:who_id).uniq).pluck(:id, :email).map { |id, email| [id, email] }]
-    jurisdiction_ids = [transfers.map(&:from_jurisdiction_id), transfers.map(&:to_jurisdiction_id)].flatten.uniq
-    jurisdiction_paths = Hash[Jurisdiction.find(jurisdiction_ids).pluck(:id, :path).map { |id, path| [id, path] }]
-    transfers.map { |transfer| transfer.custom_details(fields, patients_identifiers[transfer.patient_id], user_emails, jurisdiction_paths) }
-  end
-
-  # Extracts history data values given relevant fields
-  def extract_histories_details(patients_identifiers, histories, fields)
-    histories.map { |history| history.custom_details(fields, patients_identifiers[history.patient_id]) }
-  end
+    selected_fields = fields & %i[id patient_id created_at updated_at]
+    plucked_fields = selected_fields.dup
 
-  # Writes export data to file(s)
-  def write_export_data_to_files(config, patients_group, index, inner_batch_size)
-    case config[:format]
-    when 'csv'
-      csv_export(config, patients_group, index, inner_batch_size)
-    when 'xlsx'
-      xlsx_export(config, patients_group, index, inner_batch_size)
+    if fields.include?(:who)
+      transfers = transfers.joins('INNER JOIN users ON transfers.who_id = users.id')
+      selected_fields << 'users.email AS who'
+      plucked_fields << :who
     end
-  end
-
-  # Creates a list of csv files from exported data
-  def csv_export(config, patients_group, index, inner_batch_size)
-    # Get all of the field data based on the config
-    field_data = get_field_data(patients_group, config)
 
-    files = []
-    csvs = {}
-    packages = {}
-
-    # 1) Generate CSVs and packages for each data type
-    CUSTOM_EXPORT_OPTIONS.each_key do |data_type|
-      next unless config.dig(:data, data_type, :checked).present?
-
-      # Create CSV with column headers
-      package = CSV.generate(headers: true) do |csv|
-        csv << field_data.dig(data_type, :headers)
-        csvs[data_type] = csv
-      end
-      packages[data_type] = package
+    if fields.include?(:from_jurisdiction)
```

Comment:
I wonder if it'd be faster to eager load these associations rather than to potentially do two joins with the jurisdictions table. Could it be faster just to eager load both the jurisdiction and user associations? https://guides.rubyonrails.org/active_record_querying.html#eager-loading-multiple-associations

Reply:
not sure if that's possible because ActiveRecord does not know how to do the joins/eager loading by calling stuff like `transfers.join(:jurisdiction)` or `transfers.includes(:jurisdiction)` because the foreign key is not simply `jurisdiction_id`, there's 2 foreign keys linking to the same association separately.